src/transformState.js

```diff
@@ -5,7 +5,27 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    const { type, extraData, keysToRemove } = action;
+
+    switch (true) {
+      case type === 'addProperties' :
+        Object.assign(state, extraData);
+        break;
+      case type === 'removeProperties' :
+        for (const key of keysToRemove) {
+          delete state[key];
+        }
+        break;
+      case type === 'clear' :
+        for (const key in state) {
+          delete state[key];
+        }
+        break;
+    }
```

Comment:
Don't forget to add a default case to handle potential errors in case type doesn't match any of cases